pages/reference/OS/advanced.md

```diff
@@ -5,94 +5,91 @@ excerpts: Configuration options to expose more device functionality to {{ $names
+
+__Note:__ In addition to the dashboard, these configuration variables can be also be set using the API or any of its clients, including the [SDK][sdk] and [CLI][cli].
 
 ### GPU Memory
 
-You can configure the amount of memory shared with the GPU by adding entries to
-`config.txt` as follows:
+The [amount of memory][gpu-memory] that is addressable from the GPU my be configured by adding entries to `config.txt`. You can target different memory sizes, specified in MB, of 256, 512, and 1024 corresponding to the RAM on the Raspberry Pi. Values will be ignored for those devices whose memory size does not match. For the Raspberry Pi 4, which has versions with RAM greater than 1GB, the minimum and maximum values are the same as for a 1GB device.
 
 ```
-gpu_mem=16
 gpu_mem_256=64
 gpu_mem_512=128
+gpu_mem_1024=256
 ```
 
-Each of these values is specified in megabytes.
+`gpu_mem` is the default amount of memory, which, by default, is set to 16MB (specified by `gpu_mem=16`). This may well be less than you require depending on your application (particularly applications that make heavy use of the Raspberry Pi's graphics card). As per the [Raspberry Pi documentation][gpu-memory], values of `gpu_mem` over 512 are not recommended, will provide no performance improvements, and are untested.
 
-Note that `gpu_mem_256` is used by the 256MB Raspberry Pi (Model A - you
-probably *don't* have one of these), and `gpu_mem_512` is used by the 512MB
-Raspberry Pi (Model B/B+ - this is probably what you have.)
+### Enable serial interface
 
-The Model A Raspberry Pi ignores the 512MB setting altogether while the model B
-and B+ ignore the 256MB setting. Both `gpu_mem_256` and `gpu_mem_512` override
-`gpu_mem`.
+The BCM2837 on the Raspberry Pi 3 has 2 built-in [UARTs][uart] (as did its predecessors), however, to support the Bluetooth functionality the fully-featured PL011 UART was moved from the header pins to the Bluetooth chip and the mini UART made available on header pins 8 & 10.
 
-By default we assign 16MB of memory to the GPU (specified by `gpu_mem`.) This
-may well be less than you require depending on your application (particularly
-applications which make heavy use of the Pi's graphics card.)
+This has a number of consequences for users of the serial interface:
 
-### Enable serial interface
+- The /dev/ttyAMA0 previously used to access the UART now connects to Bluetooth.
+- The mini UART is now available on /dev/ttyS0, disabled by default.
+- The mini UART is a secondary low throughput UART intended to be used as a console. it supports the following functionality:
+  - 7 or 8 bit operation.
+  - 1 start and 1 stop bit.
+  - No parities.
+  - Break generation.
+  - 8 symbols deep FIFOs for receive and transmit.
+  - SW controlled RTS, SW readable CTS.
+  - Auto flow control with programmable FIFO level.
+  - 16550 like registers.
+  - Baudrate derived from system clock.
+
+To enable the mini UART, an entry should be added to `config.txt` as follows:
```

Comment:
This may raise questions about the fact we have a toggle for this in device configuration - how does that toggle relate to this setting?

Reply:
I'll update this for clarity but essentially it's the same as other config options we have prepopulated i.e. if you edit this before device provisioning it'll be respected. After device provisioning, it has to be done via the dashboard. Tested this by enabling UART `enable_uart=1` on a production image prior to first boot and it worked.

This also applies to Rahul's comment as for dev images this is already enabled.